### PR TITLE
added return type all in from_ * modules

### LINKED ISSUE #4082: config from_* methods should document their return type
They do return *something*, but it's not obvious what it is, so having a `:return: whatever they return` in the docstring would be nice to have.

### PR DESCRIPTION
**Pull request by starlightknown**
added return type in all `from_*` modules as discussed in the PR. 
(closed the other PR because of few mistakes which got solved in discord)

- fixes #4082 

Checklist:

- [ ] Add tests that demonstrate the correct behavior of the change. Tests should fail without the change.
- [ ] Add or update relevant docs, in the docs folder and in code.
- [ ] Add an entry in `CHANGES.rst` summarizing the change and linking to the issue.
- [ ] Add `.. versionchanged::` entries in any relevant code docs.
- [ ] Run `pre-commit` hooks and fix any issues.
- [ ] Run `pytest` and `tox`, no tests failed.

### CODE REVIEW

File: src/flask/config.py [hunk 1]
```diff
@@ -107,7 +107,7 @@ def from_pyfile(self, filename: str, silent: bool = False) -> bool:
                          root path.
         :param silent: set to ``True`` if you want silent failure for missing
                        files.
-
+        :return: bool.``True`` if able to update values in config, ``False`` otherwise.
```
Comment (pgjones) [inline src/flask/config.py:110]: I think the read the docs CI build requires a blank line after this one to pass.

**Review by pgjones — CHANGES_REQUESTED** [review]

File: src/flask/config.py (hunk 1 above)
Comment (davidism) [reply src/flask/config.py:110]: Also, remove the leading "bool.", and the trailing ", ``False`` otherwise``, they're both redundant.

Comment (starlightknown) [reply src/flask/config.py:110]: As in `from_pyfile`?

Comment (davidism) [reply src/flask/config.py:110]: "True if values were loaded successfully." sounds right for all the methods.

Comment (starlightknown) [reply src/flask/config.py:110]: Sure will update it

**Review by pgjones — APPROVED** [review]

**Comment by pgjones** [comment]
I've changed the target to the 2.0.x branch (as it would be useful in 2.0.2), could you rebase your branch and squash to a single commit?

`git rebase -i origin/2.0.x` then `git push -f`.

**Comment by starlightknown** [comment]
Sure

**Comment by starlightknown** [comment]
I keep getting merge conflicts whenever I do rebase in `__init__.py` and `CHANGES.rst` I have solved it three times now but it keeps happening again

**Comment by starlightknown** [comment]
done, I guess it's fine now?

**Comment by davidism** [comment]
Please squash this into a single commit.

**Pull request merged**